### PR TITLE
Default to `ltr` for reading direction

### PR DESCRIPTION
**Pull request by farmio**
## Proposed change

- Default to `ltr` for reading direction of icons instead of `rtl`
- Use `mainWindow`s `document.dir` to get correct reading direction even from iFrames

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/frontend/pull/18087#issuecomment-1820479727
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-icon-button-arrow-next.ts [hunk 1]
```diff
@@ -13,7 +14,17 @@ export class HaIconButtonArrowNext extends LitElement {
...
-    document.dir === "ltr" ? mdiArrowRight : mdiArrowLeft;
+    document.dir === "rtl" ? mdiArrowLeft : mdiArrowRight;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    if (!document.dir) {
+      const dir = this.hass ? computeRTLDirection(this.hass) : undefined;
+      if (dir === "rtl") {
+        this._icon = mdiArrowLeft;
+      }
```
Comment (bramkragten) [inline src/components/ha-icon-button-arrow-next.ts:25]:
```suggestion
	if (this.hass) {
          this._icon = computeRTLDirection(this.hass) === "rtl" ? mdiArrowLeft : mdiArrowRight;
        }
```

Comment (bramkragten) [inline src/components/ha-icon-button-arrow-next.ts:25]:
```suggestion
    if (!document.dir && this.hass) {
        this._icon = computeRTLDirection(this.hass) === "rtl" ? mdiArrowLeft : mdiArrowRight;
      }
```

**Comment by bramkragten** [comment]
You can use [mainWindow](https://github.com/home-assistant/frontend/blob/dev/src/common/dom/get_main_window.ts#L3), this will always give the window of home assistant, also in iFrames.

```ts
mainWindow.document.dir === "rtl" ? mdiArrowLeft : mdiArrowRight;
```

**Comment by farmio** [comment]
Great! I think we don't need any `connectedCallback()` at all then, do we?
I did find 4 more occasions of `document.ltr` that I added to this PR
- ha-fab
- ha-button-menu
- ha-textarea
- ha-textfield

**Review by bramkragten — APPROVED** [review]

**Pull request merged**